sdk/core/core-http/src/policies/systemErrorRetryPolicy.ts

```diff
@@ -88,7 +88,7 @@ export class SystemErrorRetryPolicy extends BaseRequestPolicy {
   public sendRequest(request: WebResourceLike): Promise<HttpOperationResponse> {
     return this._nextPolicy
       .sendRequest(request.clone())
-      .then((response) => retry(this, request, response));
+      .catch((error) => retry(this, request, error.response, undefined, error))
```

Comment:
minor nit: could we reorder so that retryData comes last to avoid having to pass undefined?

Reply:
Good suggestion! Updated.